pkg/clusteragent/externalmetrics/datadogmetric_controller.go

```diff
@@ -208,8 +208,10 @@ func (c *DatadogMetricController) processDatadogMetric(key interface{}) error {
 		if datadogMetricCached != nil {
 			// Feeding local cache with DatadogMetric information
 			c.store.Set(datadogMetricKey, model.NewDatadogMetricInternal(datadogMetricKey, *datadogMetricCached), ddmControllerStoreID)
+			setDatadogMetricTelemetry(datadogMetricCached)
 		} else {
 			c.store.Delete(datadogMetricKey, ddmControllerStoreID)
+			unsetDatadogMetricTelemetry(ns, name)
```

Comment:
This is the follower's path, don't we want only the leader to update this metric?

Reply:
The telemetry also uses `le.JoinLeaderValue` to allow querying for the value from just the leader. Having the followers with up to date telemetry at all times also prevents a brief period of inaccurate data when a new leader is elected, since we don't have to wait for a resync to update it.